### ISSUE TITLE
Adam Audit issue #1: withdrawEnableStarttime of user X can be reset by anyone  by transfering 0 tokens using `transferFrom(X, to, 0)`

### ISSUE THREAD

**Issue by jellegerbrandy** [hats-finance/hats-contracts#366]
adam's comments:
1) this check: https://github.com/hats-finance/hats-contracts/blob/1272ef58df5d65fcc8d4cddbb23772cd4a6a5cf3/contracts/HATVault.sol#L498 can be bypassed by doing a zero token transfer, which will always succeed and will reset withdrawEnableStartTime[from] to 0
see:
https://github.com/hats-finance/hats-contracts/blob/1272ef58df5d65fcc8d4cddbb23772cd4a6a5cf3/contracts/HATVault.sol#L725
-> suggests to remove the check in https://github.com/hats-finance/hats-contracts/blob/1272ef58df5d65fcc8d4cddbb23772cd4a6a5cf3/contracts/HATVault.sol#L498

**Comment by jellegerbrandy** [comment]
I do not see how this suggestion fixes the underlying problem. The problem is that if I call `transferFrom(victim, anyAddress, 0)`, this will succeed, and set victim's withdrawenablestarttime to 0. Right?

**Comment by jellegerbrandy** [comment]
Is this issue still valid after fixing https://github.com/hats-finance/hats-contracts/issues/367 ?

In the current logic, you cannot transfer any tokens away from your account if there is no active withdrawrequest. After fixing #367 , no transferFrom(from,...) will work if from does not have an active withdraw request. 

So the check cannot be bypassed anymore. Do we still want to remove the check? REmoving the check is not only a UX issue, it does give the user more power on scheduling here withdrawrequests. Which we may not want to allow bc we do not want the user to play too much with the withdrawals (although I do not have a real attack)

**Comment by lirona** [comment]
> Is this issue still valid after fixing #367 ?
> 
> In the current logic, you cannot transfer any tokens away from your account if there is no active withdrawrequest. After fixing #367 , no transferFrom(from,...) will work if from does not have an active withdraw request.
> 
> So the check cannot be bypassed anymore. Do we still want to remove the check? REmoving the check is not only a UX issue, it does give the user more power on scheduling here withdrawrequests. Which we may not want to allow bc we do not want the user to play too much with the withdrawals (although I do not have a real attack)

i agree, if a user already has a pending withdraw request there is no reason he will need to request again, it only prolongs the time he needs to wait to withdraw, so that could only be if he's trying to manipulate to have a certain time in the more distant future to withdraw before the withdraw enable period will end. although a user can deposit/withdraw a small amount and reset the withdraw request at any time so maybe it doesn't matter.

**Comment by ben-kaufman** [comment]
It doesn't give the user any extra power, since the user could cancel the request by calling deposit, then making a new request. So in terms of user power nothing changes. This is only a UI question, and we normally say that issues of the UI should be resolved in the UI, not the contracts, that's why I think we can remove that. But really not so important...

**Comment by jellegerbrandy** [comment]
ok let's remove it :)
